pkg/sql/conn_executor_exec.go

```diff
@@ -244,7 +244,10 @@ func (ex *connExecutor) execStmtInOpenState(
 			ex.incrementExecutedStmtCounter(stmt)
 		}
 	}()
+
+	ex.state.mu.Lock()
 	ex.state.mu.stmtCount++
```

Comment:
I wonder if it's worth using atomic increments and loads for this one

Reply:
Yeah I suppose that would be better, but I think lock contention will be quite rare here so I'm going to go ahead and merge as-is for now since this is affecting multiple tests.